app/controllers/application_controller.rb

```diff
@@ -28,13 +28,21 @@ def user_signed_in?
   end
 
   def after_sign_in_path
-    if current_user.support_user?
+    if requested_path.present?
+      requested_path
+    elsif current_user.support_user?
       support_root_path
     else
       organisations_path
     end
   end
 
+  def requested_path
+    return if [sign_in_path, sign_out_path, nil].include?(session["requested_path"])
+
+    session[:requested_path]
+  end
```

Comment:
We should be careful to make sure we [always use strings for session keys][1]. Symbols will probably work just fine, but it's a bit weird because different session adapters behave differently. AFAIK every adapter works as expected with strings.

[1]: https://github.com/DFE-Digital/itt-mentor-services/commit/3f37281c7560bdb031b5baf5db73353dd7c3292a

```suggestion
  def requested_path
    return if [sign_in_path, sign_out_path, nil].include?(session["requested_path"])

    session["requested_path"]
  end
```